core/modules/layout/includes/custom_text_block.inc

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/**
+ *  CustomTextBlock extends Block
+ *  
+ *  This class allows us to create custom text blocks.
+ */
+class CustomTextBlock extends Block {
+  /**
+   *  Sets title text on draggable block panel in Layout builder.
+   */
+  function getAdminTitle() {
+    return t('Custom text block');
+  }
+
+  /**
+   *  Sets description text on draggable block in Layout builder.
+   */
+  function getAdminPreview() {
+    $label_text = !empty($this->settings['subject']) ? ': ' . check_plain($this->settings['subject']) : '';
+    $preview = '<p>Custom text' . $label_text;
+    $preview .= '</p>';
+    
+    if (!empty($this->conditions)) {
+      foreach ($this->conditions as $condition) {
+        $condition_list[] = $condition->summary();
+      }
+      $conditions = theme('item_list', array('items' => $condition_list, 'attributes' => array('class' => array('block-conditions'))));
+      $fieldset = array(
+        '#type' => 'fieldset',
+        '#collapsible' => TRUE,
+        '#collapsed' => TRUE,
+        '#title' => t('Conditions'),
+        '#children' => $conditions,
+        '#parents' => array(),
+      );
+      $form_state = array();
+      form_process_fieldset($fieldset, $form_state);
+      $preview .= backdrop_render($fieldset);
+    }
+
+    return $preview;
+  }
+
+  /**
+   *  Sets block subject on block view.
+   */
+  function getTitle() {
+    return isset($this->settings['subject']) ? check_plain($this->settings['subject']) : '';
+  }
+
+  /**
+   *  Sets block content on block view.
+   */
+  function getContent() {
+    if (!empty($this->settings['content'])) {
+      return check_markup(token_replace($this->settings['content']), $this->settings['format'], '', TRUE);
+    }
+    return 'No content saved';
```

Comment:
We should translate "No content saved". But is this going out to the front-end of the user's site? I'm not sure we should special-case anything at all here. Or maybe we only show this message to users that have "administer layouts" permission so that they can actually do something about it.